### DISCUSSION TITLE
Retopology training

### DISCUSSION THREAD

**Discussion by drperpen** [pyg-team/pytorch_geometric#8883] [Q&A]
Hi,
my goal is to train a model that can perform automatic re-topology on 3D meshes. The aim is to transform irregular, non-topologically consistent meshes into regular, topologically consistent ones. The result will always have the same point count. I would appreciate any tips on what strategy should I follow to annotate the dataset for training. As an example, below I would need to establish a relationship between mesh A on the left (irregular) and mesh B on the right (regular) in order to train:
![pyg_meshes](https://github.com/pyg-team/pytorch_geometric/assets/51986483/22a33c6a-ac8d-49db-955e-a2168c2973ff)

Thanks!

**Comment by rusty1s** [comment]
I am not totally sure if a neural network is the best approach to tackle this. Feels like some fixed function algorithm might perform better here. It will be hard to encode all the constraints you have on the output into the model IMO.

**Reply by drperpen** [reply]
Thanks for your reply! I was trying to find a way to train a dataset (which I already have) in the way they did here:
https://github.com/tianyeli/tofu

Unfortunately they didn't release any training code and in the paper they just give a high-level approach to it.
